src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -73,6 +73,20 @@ CHIP_ERROR CASEServer::OnUnsolicitedMessageReceived(const PayloadHeader & payloa
 CHIP_ERROR CASEServer::OnMessageReceived(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
                                          System::PacketBufferHandle && payload)
 {
+    if (mDoingCASEHandshake)
+    {
+        if (GetSession().IsHandleSigma3PermanentlyBlockedOnBackgroundWork())
+        {
+            ChipLogError(Inet, "CASE session helper failed to schedule HandleSigma3c, executing HandleSigma3c");
+            GetSession().UnblockHandleSigma3FromForegroundWork();
+        }
+        else
+        {
+            ChipLogError(Inet, "CASE session is in establishing state, returning without responding");
```

Comment:
Please add a comment pointing to https://github.com/project-chip/connectedhomeip/issues/27473